File: js/problems/169-majority-element.js
// const nums = [3,2,3];
const nums = [1];

/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
  let score = {};
  let max = 0;

  for (let num of nums) {
      if (score[num]) {
          score[num] = score[num] + 1;
      } else {
          score[num] = 1;
      }

      if (score[num] > max) max = score[num];
  }

  return Object.keys(score).find(key => score[key] === max);
};

majorityElement(nums);

File: cpp/problems/169-majority-element.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
 public:
  int majorityElement(vector<int>& nums) {
    unordered_map<int, int> hash;
    int maxKey = 0;
    int maxValue = 0;

    for (int num : nums) {
      hash[num] = 1 + hash[num];

      if (hash[num] > maxValue) {
        maxKey = num;
        maxValue = hash[num];
      }
    }

    return maxKey;
  }
};

int main() {
  vector<int> v = {3, 2, 3};

  Solution solution;
  solution.majorityElement(v);

  return 0;
}
